### PR TITLE
feat:re-run logic, test more than once incase of failure (Outdated will be closed)

### PR DESCRIPTION
**Pull request by jazibhumayun**
Re-running upon failed tests, before notifying it as failure.

### CODE REVIEW

File: dataeng/resources/warehouse-transforms-dbt.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+# marking a job as failed job
+
+cd $WORKSPACE/warehouse-transforms/projects/$DBT_PROJECT_PATH
+
+dbt clean --profiles-dir $WORKSPACE/analytics-secure/warehouse-transforms/ --profile $DBT_PROFILE --target $DBT_TARGET
+dbt deps --profiles-dir $WORKSPACE/analytics-secure/warehouse-transforms/ --profile $DBT_PROFILE --target $DBT_TARGET
+dbt seed --profiles-dir $WORKSPACE/analytics-secure/warehouse-transforms/ --profile $DBT_PROFILE --target $DBT_TARGET
+
+dbt run $DBT_RUN_OPTIONS $DBT_RUN_EXCLUDE --profiles-dir $WORKSPACE/analytics-secure/warehouse-transforms/ --profile $DBT_PROFILE --target $DBT_TARGET
+# NO_OF_TRIES is currently set to '2' which will execute dbt test '2' times and can be increase as per the need.
```
Comment (jazibhumayun) [inline dataeng/resources/warehouse-transforms-dbt.sh:16]: re-run logic below

File: dataeng/resources/warehouse-transforms-dbt.sh [hunk 2]
```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env bash
```
Comment (macdiesel) [inline dataeng/resources/warehouse-transforms-dbt.sh:1]: Lets add -with-retry to the name of the script.

File: dataeng/resources/warehouse-transforms-dbt.sh [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+cd $WORKSPACE/warehouse-transforms/projects/$DBT_PROJECT_PATH
+
+dbt clean --profiles-dir $WORKSPACE/analytics-secure/warehouse-transforms/ --profile $DBT_PROFILE --target $DBT_TARGET
+dbt deps --profiles-dir $WORKSPACE/analytics-secure/warehouse-transforms/ --profile $DBT_PROFILE --target $DBT_TARGET
+dbt seed --profiles-dir $WORKSPACE/analytics-secure/warehouse-transforms/ --profile $DBT_PROFILE --target $DBT_TARGET
+
+dbt run $DBT_RUN_OPTIONS $DBT_RUN_EXCLUDE --profiles-dir $WORKSPACE/analytics-secure/warehouse-transforms/ --profile $DBT_PROFILE --target $DBT_TARGET
+# NO_OF_TRIES is currently set to '2' which will execute dbt test '2' times and can be increase as per the need.
+NO_OF_TRIES=2
```
Comment (macdiesel) [inline dataeng/resources/warehouse-transforms-dbt.sh:17]: Lets set this to 3 to start.

**Review by macdiesel — CHANGES_REQUESTED** [review]
Just a few requests.

**Comment by jazibhumayun** [comment]
Created a new PR #1353 that is using ci_script to re-run flaky tests

**Comment by jazibhumayun** [comment]
This PR will be closed and no longer will be needed.

**Pull request closed without merging**